lib/CTownHandler.cpp

```diff
@@ -343,28 +343,93 @@ void CTownHandler::loadBuildingRequirements(CBuilding * building, const JsonNode
 	requirementsToLoad.push_back(hlp);
 }
 
+template<typename R>
+R CTownHandler::getMappedValue(const std::string key, const R defval, const std::map<std::string, R> & map) const
+{
+	for(const auto & kvp : map)

```

Comment:
You should use `map::find()` here. Also there is an almost identical function `parseByMap` in JsonNode.cpp